src/NUnitFramework/framework/Internal/AwaitAdapter.cs

```diff
@@ -37,8 +37,7 @@ public static AwaitAdapter FromAwaitable(object awaitable)
             // special quality of blocking until complete in GetResult.
             // As long as the pattern-based adapters are reflection-based, this
             // is much more efficient as well.
-            var task = awaitable as System.Threading.Tasks.Task;
-            if (task != null) return TaskAwaitAdapter.Create(task);
+            if (awaitable is System.Threading.Tasks.Task task) return TaskAwaitAdapter.Create(task);
```

Comment:
I would prefer the `return` statement to be on the next line, but that is outside your scope.

Reply:
I agree with @manfred-brands. I sometimes overlook `return´ statements when they are on the same line.

Reply:
Thanks! Makes sense to me. Updated